domain/application/charm/store/store.go

```diff
@@ -66,3 +67,17 @@ func (s *CharmStore) Store(ctx context.Context, name string, path string, size i
 	// Store the file in the object store.
 	return objectStore.PutAndCheckHash(ctx, uniqueName, file, size, hash)
 }
+
+// Get retrieves a ReadCloser for the charm archive at the give path from
+// the underlying storage.
```

Comment:
Can we add a note, that the integrity of the file (the hash) is on the user/client to verify. Once stored in the system, we **assume** that file hasn't been changed once written.

Reply:
Should we verify this in the service layer do you think?

Reply:
We can do that asynchronously if it's a problem, i.e. in a worker.